### ISSUE TITLE
MIssing cmdlet "ConvertFrom-Yaml"

### ISSUE THREAD

**Issue by Gill-Bates** [PowerShell/PowerShell#24654]
### Summary of the new feature / enhancement

Missing painfully an option to Convert a YAML to a Powershell Object like

`ConvertFrom-Yaml
`

Right now, I must use a Workaround: https://www.powershellgallery.com/packages?q=powershell-yaml

Almost 100 Mio Downloads should speak for itself.

THanks for taking this into consideration.

### Proposed technical implementation details (optional)

https://www.powershellgallery.com/packages?q=powershell-yaml

**Comment by rhubarb-geek-nz** [comment]
> Almost 100 Mio Downloads should speak for itself.

Indeed, it shows that [PSGallery](https://www.powershellgallery.com/) is an excellent place to host cmdlets to extend PowerShell.

**Comment by kilasuit** [comment]
This has been asked for previously in #3607 #16785 #16819 & possibly more too and ultimately has been declined each time to be added because we have a number of packages on the Gallery that provides this functionality already, with no .NET native YAML functionality.

Some additional background conversations in dotnet repos https://github.com/dotnet/runtime/issues/83313 & https://github.com/dotnet/runtime/issues/109199

We've also had discussions around providing a community driven "DevOps" meta-module on the gallery, that works similar to the Az/Microsoft.Graph module for a download once requirement and would include a module like PowerShell-Yaml along with a number of others.

I am tempted in marking this as resoltion answered & duplicate, but will leave doing so right now as to allow for other comments, however this would be for the cmdlets WG to discuss so marking it for under that WG & it may be re-discussed in the New Year

**Comment by kasini3000** [comment]
I suggest building it in again.

**Comment by rhubarb-geek-nz** [comment]
This appears to be the original repository for [powershell-yaml](https://github.com/cloudbase/powershell-yaml)

As you can see by the number of forks(78) and the [outstanding issues](https://github.com/cloudbase/powershell-yaml/issues) there is no right solution for how to do Yaml with PowerShell, should it be hashtables or PSCustomObject, how should numbers be parsed etc.

I would far rather those discussions play out in the specific module than the PowerShell project having to decide on the one true parser or myriad of options to cater for everyone's requirements.

As an installable module it can already be used with any version of PowerShell later than 3.0.

**Comment by StephenHodgson** [comment]
> I am tempted in marking this as resoltion answered & duplicate

You'd think with all the links and references where the community asked for first party support, I'd have enough momentum to become reality 😏

**Comment by iRon7** [comment]
> @kilasuit With this reasoning, all cmdlets that contain ConvertFrom/To would have to disappear from Powershell.
> 
> I plead for a built-in solution that is tested and secure.

To my opinion `ConvertFrom`/`To` cmdlets shouldn't actually exist knowing that they aren't based on an approved verb.
It would have been better if there was a more general cmdlet that contains common serialization/de-serialization logic along with error messages etc. which should be easier to maintain. The cmdlet would than be called "`Convert-Object`" with parameter sets containing mandatory parameters as:
* `-ToJson` / `-FromJson`
* `-ToYaml` / `-FromYaml`
* `-ToCsv` / `-FromCsv`
* `-ToXml`
* `-FromString`
* ...

I think it is even not to late for such a direction.
A set of `ConvertFrom`/`To` proxy cmdlets might than be made available (through e.g. a "The kitchen sink" edition) for downwards compatibility.

**Comment by rhubarb-geek-nz** [comment]
> The cmdlet would than be called "`Convert-Object`" with parameter sets containing mandatory parameters ..

I think that would be the worst of all possible solutions, it means new format types can't be added without a new release of PowerShell.

I suggest that you can have a Convert-Object command but it just has "-To" and "-From" parameters which accept the desired format. Then the command can internally invoke the appropriate ConvertTo-XXXX and ConvertFrom-XXXX without any knowledge of the formats and leaving the system extensible using the the existing ConvertTo/ConvertFrom pattern.

eg 

```
Convert-Object -To CSV
```

or

```
Convert-Object -From JSON
```

Which would delegate respectively to `ConvertTo-CSV` and` ConvertFrom-Json`

**Comment by iRon7** [comment]
> Both ConvertFrom and ConvertTo _are_ [PowerShell approved verbs](https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.4#data-verbs).

Oops, you right in every respect. I simply forgot about that. I could have split the comment in two general `ConvertTo-Object` and  `ConvertFrom-Object` but that would be even worse. I decided to delete the entry.

**Comment by SteveL-MSFT** [comment]
The @PowerShell/wg-powershell-cmdlets discussed this.  We understand and agree that YAML is used heavily in the industry and would also like to see it supported natively within PS7.  Adding a new dependency on an external assembly will introduce dependency collisions.  We've already experienced such issues in the past.  We also recognize that the popularity of the YAML modules on PSGallery means that YAML support is needed, but also that many folks are ok using 3rd party modules for YAML.  If YAML support shows up in .NET runtime, we would certainly approve of builtin YAML cmdlets in PS7.
